src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Components/TypeNameHelper.cs

```diff
@@ -79,6 +79,12 @@ public static void WriteGloballyQualifiedName(CodeWriter codeWriter, string type
     }
 
     internal static void WriteGloballyQualifiedName(CodeWriter codeWriter, ReadOnlyMemory<char> typeName)
+    {
+        WriteGlobalQualifierNameIfNeeded(codeWriter, typeName);
+        codeWriter.Write(typeName);
+    }
+
+    internal static void WriteGlobalQualifierNameIfNeeded(CodeWriter codeWriter, ReadOnlyMemory<char> typeName)
```

Comment:
This method name confused me for a bit since there's no "name" being written. Consider something like, 'WriteGlobalPrefixIfNeeded` and add a doc comment to describe that this method writes `global:::` if it isn't already included at the start of the type name.